cmd/cli/plugin/airgapped-network/publishimagestotar.go

```diff
@@ -0,0 +1,240 @@
+// Copyright 2022 VMware, Inc. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package main
+
+import (
+	"fmt"
+	"os"
+	"path"
+	"path/filepath"
+	"sort"
+	"strconv"
+	"strings"
+
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+	"sigs.k8s.io/yaml"
+
+	tkrv1 "github.com/vmware-tanzu/tanzu-framework/apis/run/pkg/tkr/v1"
+)
+
+var imageinfo = map[string]string{}
+
+type publishimagestotarOptions struct {
+	tkgImageRepo    string
+	tkgVersion      string
+	tarFilePath     string
+	customImageRepo string
+}
+
+var pullImage = &publishimagestotarOptions{}
+
+var publishImagestotarCmd = &cobra.Command{
+	Use:          "publish-image-to-tar",
+	Short:        "Copy images from public repo to tar files",
+	RunE:         publishimagestotar,
+	SilenceUsage: true,
+}
+
+func init() {
+	publishImagestotarCmd.Flags().StringVarP(&pullImage.tkgImageRepo, "tkgImageRepository", "", "projects.registry.vmware.com/tkg", "TKG public repository")
+	publishImagestotarCmd.Flags().StringVarP(&pullImage.tkgVersion, "tkgVersion", "", "", "TKG version")
+	publishImagestotarCmd.Flags().StringVarP(&pullImage.customImageRepo, "customImageRepo", "", "", "custom images repository for airgapped network")
+}
+
+func downloadTkgCompatibilityImage() {
+	fmt.Printf("--- start the process of tkg-compatibility ---")
+	tkgCompatibilityRelativeImagePath := "tkg-compatibility"
+	tkgCompatibilityImagePath := path.Join(pullImage.tkgImageRepo, tkgCompatibilityRelativeImagePath)
+	imageTags := imgpkgTagListImage(tkgCompatibilityImagePath)
+	sort.SliceStable(imageTags, func(i, j int) bool {
+		vi, err := strconv.Atoi(strings.TrimPrefix(imageTags[i], "v"))
+		if err != nil {
+			printErrorAndExit(errors.Wrapf(err, "parse tkg-compatibility image tag %s failed", imageTags[i]))
+		}
+		vj, err := strconv.Atoi(strings.TrimPrefix(imageTags[j], "v"))
```

Comment:
The code to sort is repeated in other function as well. Is it possible to remove duplication?